egapro/db.py

```diff
@@ -52,6 +52,12 @@ async def all(cls):
         # TODO ORDER BY ?
         return await cls.fetch("SELECT * FROM declaration")
 
+    @classmethod
+    async def all_by_siren(cls, siren):
+        return await cls.fetch(
+            "SELECT * FROM declaration WHERE siren=$1 ORDER BY year DESC", siren
```

Comment:
`SELECT *` without a limit is 💣

Reply:
What limit would you set by default? Would you also implement that feature in [other functions](https://github.com/SocialGouv/egapro-api/pull/6/files/bdb0baedec71641d9a6e1a66e5ed95d9d830b11d#diff-dd90d84ad412dd90a59c92b2514dc3a8R53)?

Reply:
I would put a documented high limit (like 10000? depends on the data fetched?), better for a developer to spend some time finding why than for users to reach a broken website at some point.

Maybe a global reusable var on that file? @yohanboniface ?

Reply:
By the way, we may only select the wanted property:

```
SELECT data->'declaration'->'noteIndex' as index
```

About the limit, I'm not sure this will be a real problem, given we filter by siren and there is unique constraint on the couple year/siren.

Reply:
Would it make sense for a generic function `all_by_siren`?
Or are you suggesting to make a very specific function `indexes_by_siren`? In the latest case we should also retrieve the year besides.